utils/eth1.go

```diff
@@ -309,103 +310,117 @@ func FormatAddressLong(address string) template.HTML {
 
 }
 
-func FormatAmountFormatted(amount *big.Int, unit string, digits int, maxPreCommaDigitsBeforeTrim int, fullAmountTooltip bool, smallUnit bool, newLineForUnit bool) template.HTML {
-	return formatAmount(amount, unit, digits, maxPreCommaDigitsBeforeTrim, fullAmountTooltip, smallUnit, newLineForUnit)
+func FormatAmountFormatted(amountInWei *big.Int, targetCurrency string, maxFractionalDigitsBeforeTrim int, maxIntegerDigitsBeforeTrim int, fullAmountTooltip bool, smallUnit bool, newLineForUnit bool) template.HTML {
+	return convertAndFormatWei(amountInWei, targetCurrency, maxFractionalDigitsBeforeTrim, maxIntegerDigitsBeforeTrim, fullAmountTooltip, smallUnit, newLineForUnit)
 }
-func FormatAmount(amount *big.Int, unit string, digits int) template.HTML {
-	return formatAmount(amount, unit, digits, 0, true, false, false)
+func FormatAmount(amountInWei *big.Int, targetCurrency string, maxFractionalDigitsBeforeTrim int) template.HTML {
+	return convertAndFormatWei(amountInWei, targetCurrency, maxFractionalDigitsBeforeTrim, 0, true, false, false)
 }
-func FormatBigAmount(amount *hexutil.Big, unit string, digits int) template.HTML {
-	return FormatAmount((*big.Int)(amount), unit, digits)
+func FormatBigAmount(amountInWei *hexutil.Big, targetCurrency string, maxFractionalDigitsBeforeTrim int) template.HTML {
+	return FormatAmount((*big.Int)(amountInWei), targetCurrency, maxFractionalDigitsBeforeTrim)
 }
-func FormatBytesAmount(amount []byte, unit string, digits int) template.HTML {
-	return FormatAmount(new(big.Int).SetBytes(amount), unit, digits)
+func FormatBytesAmount(amountInWei []byte, targetCurrency string, maxFractionalDigitsBeforeTrim int) template.HTML {
+	return FormatAmount(new(big.Int).SetBytes(amountInWei), targetCurrency, maxFractionalDigitsBeforeTrim)
 }
-func formatAmount(amount *big.Int, unit string, digits int, maxPreCommaDigitsBeforeTrim int, fullAmountTooltip bool, smallUnit bool, newLineForUnit bool) template.HTML {
-	// define display unit & digits used per unit max
-	displayUnit := " " + unit
-	var unitDigits int
-	if unit == "ETH" || unit == "Ether" || unit == "xDAI" || unit == "GNO" {
-		unitDigits = 18
-	} else if unit == "GWei" {
-		unitDigits = 9
-	} else {
-		displayUnit = " ?"
-		unitDigits = 0
+
+func convertAndFormatWei(amountInWei *big.Int, targetCurrency string, maxFractionalDigitsBeforeTrim int, maxIntegerDigitsBeforeTrim int, fullAmountTooltip bool, smallUnit bool, newLineForUnit bool) template.HTML {
+	var commaPositionFromEnd int
+	var targetAmount *big.Int
+
+	targetAmount = amountInWei
```

Comment:
**nitpick:** This could just be one line.

Reply:
done.